hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -220,7 +220,9 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
    *       partition path, meaning that string value of "2022/01/01" will be appended, and not its original
    *       representation
    */
-  protected val shouldExtractPartitionValuesFromPartitionPath: Boolean = {
+  protected val shouldExtractPartitionValuesFromPartitionPath: Boolean = needExtractPartitionValuesFromPartitionPath()
+
+  protected def needExtractPartitionValuesFromPartitionPath(): Boolean = {
     // Controls whether partition columns (which are the source for the partition path values) should
```

Comment:
Why add a new method name?

Reply:
Because `shouldExtractPartitionValuesFromPartitionPath` is val, which can not be override as follows:
```scala
override protected val shouldExtractPartitionValuesFromPartitionPath: Boolean = {
  if (hasSchemaOnRead) {
    super.shouldExtractPartitionValuesFromPartitionPath
  } else true
}
```